src/main/java/io/quarkus/sample/TodoAgentWebSocket.java

```diff
@@ -0,0 +1,34 @@
+package io.quarkus.sample;
+
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.subscription.MultiEmitter;
+import jakarta.enterprise.context.control.ActivateRequestContext;
+
+import io.quarkus.websockets.next.OnOpen;
+import io.quarkus.websockets.next.OnTextMessage;
+import io.quarkus.websockets.next.PathParam;
+import io.quarkus.websockets.next.WebSocket;
+
+
+@WebSocket(path = "/todo-agent/{todoId}")
+@ActivateRequestContext
+public class TodoAgentWebSocket {
+    private MultiEmitter<? super String> emitter;
+    private Multi<String> agentStream;
+
+    @OnOpen
+    Multi<String> onOpen(@PathParam String todoId) {
+        this.agentStream = Multi.createFrom().emitter(emitter -> {
+            this.emitter = emitter;
+            // The emitter is now available for use elsewhere
+        });
+        emitter.emit("ok we will find agents for todo " + todoId );
```

Comment:
Likely null, as the emitter will be created during the subscription.
So you can't emit at that point.

Reply:
Stream != Connection

You can immediately send a message back from the server to the client (just return what you want to send). 

But if ai understand correctly, you want the client to send a message with the todo id? In that case this todo is can be passed as path parameter or in a first message (handled using OnTextMessage).